### PR TITLE
feat: include coverage message

### LINKED ISSUE #101: Add Comments to PRs with Coverage Details
**Is your feature request related to a problem? Please describe.**
I really like the new feature where the lines missed will print in the output of the actions; however, I find it kinda annoying to go into the action details and click through to find what I am looking for.

**Describe the solution you'd like**
I think it would be really cool if the action could be optionally configured to write a comment on your PR with a brief report of your coverage stats. We could include if it passed, what percentage it calculated, and list any lines missed. When you re-ran the action, it would update the comment with the new details, so it wouldn't pollute the PR.

The feature is inspired by how [the firebase hosting deploy action](https://github.com/FirebaseExtended/action-hosting-deploy) adds a comment with a preview URL. I think we can also take queues [from their code](https://github.com/FirebaseExtended/action-hosting-deploy/blob/main/src/postOrUpdateComment.ts) for how to accomplish it.

I would propose adding new inputs as well:
- `report_coverage_comment` (boolean, defaults `false`): Set true if you'd like the action to comment on your PR
- `project_name` (string, optional): If set, prefixes the heading of the comment with the name (so that it's easy to identify which comment is for which project if you have multiple actions. Otherwise, defaults to the name of the workflow.

**Describe alternatives you've considered**
One alternative I've played with is adding outputs to the existing action - that way, subsequent steps could post the comment (like using [this action](https://github.com/peter-evans/create-or-update-comment)). That works well enough ([I have a fork working now](https://github.com/mtwichel/very-good-coverage)), but it really requires a fair bit of work to make your workflow work (for example, conditionally writing a new comment vs update existing; formatting with markdown, ect). I'd rather have it contained in an action to make it easy to add to lots of projects.

Furthermore, when the coverage was below the threshold, it errored and the workflow failed (as expected). However, this meant I couldn't use those outputs in subsequent steps (because the workflow terminated). I believe that I could figure out a way around that using a separate job or something like that, but that also increases the complexity of the workflow file.

Thanks as always VGV! I'm happy to help make this a reality, but I think there's some debate on the best way to set this up :)

### PR DESCRIPTION
**Pull request by alestiago**
## Description

Automatically adds a comment to the pull request with coverage information. If a comment already exists it automatically updates the comment with the new information to avoid polluting the conversation history.

Includes two new parameters:
- `report_coverage_comment` which reports coverage information as a pull request comment.
- `github_token` token required to comment on a pull request.

Aims to resolve #101 .

## Comment formats

Successful comment looks like:

## ✅ VeryGoodCoverage
Coverage: 100% (100 of 100 lines)
Min coverage difference: 0% (100% / 100%)

Unsuccessful comment looks like:

## ❌ VeryGoodCoverage
Coverage: 95% (95 of 100 lines)
Min coverage difference: -1% (95% / 96%)

<details>
<summary> Lines not covered </summary>
- /Users/felix/Development/github.com/felangel/bloc/packages/bloc/lib/src/bloc_observer.dart: 20, 27, 36, 43, 51
</details>

## Type of Change

- [X] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Review by felangel — CHANGES_REQUESTED** [review]
Can we add tests for the changes?

**Comment by alestiago** [comment]
> https://github.com/VeryGoodOpenSource/very_good_coverage/pull/159#pullrequestreview-1045580742

Are we happy with the above comment formats?

File: index.test.js [hunk 1]
```diff
@@ -124,3 +131,225 @@ test('shows lines that are missing coverage when failure occurs', () => {
...
+  const comment = 'some comment';
+  await createGitHubComment('token', comment);
+  expect(createComment).toHaveBeenCalledWith({
+    ...context.repo,
+    issue_number: context.issue.number,
+    body: comment,
+  });
+});
+
+test('formatCoverageAsMessage returns normally', () => {
```
Comment (alestiago) [inline index.test.js:340]: More test can be added to verify the formatting is correct. I'll wait until an agreed format is decided.

File: index.test.js [hunk 2]
```diff
@@ -124,3 +131,225 @@ test('shows lines that are missing coverage when failure occurs', () => {
...
+
+  try {
+  } catch (err) {
+    expect(err).toBeDefined();
+  }
+
+  // TODO(alestiago): Figure out how we want to test this, since
+  // we are mocking the github context and octokit but we seem to
+  // be unable to do so within the cp.execSync call.
+  // expect(createComment).toHaveBeenCalled();
```
Comment (alestiago) [inline index.test.js:171]: @felangel 👀

**Review by alestiago — COMMENTED** [review]
Refactored code as said in the previous comment. Designed unit tests for the new added functions. 

Missing an integration test that checks the run method acts as expected. Needs further discussion to wether we want to test with a real endpoint or we want to fully mock.

(cc: @felangel )

**Pull request closed without merging**